src/Tools/DynamoFeatureFlags/Program.cs

```diff
@@ -0,0 +1,100 @@
+﻿using CommandLine;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DynamoFeatureFlags
+{
+
+    class Options
+    {
+        [Option('u', "userkey", Required = false, HelpText = "stable user key, if not provided, a shared key will be used.")]
+        public string UserKey { get; set; }
+        [Option('m', "mobilekey", Required = false, HelpText = "mobile key for dynamo feature flag env. Do not use a full sdk key. If not provided loaded from config.")]
+        public string MobileKey { get; set; }
+
+    }
+    static class Program
+    {
+        static FeatureFlagsManager FeatureFlags { get; set; }
+        private const string checkFeatureFlagCommandToken = @"<<<<<CheckFeatureFlag>>>>>";
+        private const string endOfDataToken = @"<<<<<Eod>>>>>";
+        private const string startOfDataToken = @"<<<<<Sod>>>>>";
+        static void Main(string[] args)
+        {
+            Parser.Default.ParseArguments<Options>(args).WithParsed(ops =>
+            {
+                try
+                {
+                    FeatureFlagsManager.MessageLogged += FeatureFlagsManager_MessageLogged;
+                    FeatureFlags = new FeatureFlagsManager(ops.UserKey, ops.MobileKey);
+                    Console.WriteLine("");
+                    while (true)
+                    {
+                       
+                        var line = Console.ReadLine();
+                        if (line == checkFeatureFlagCommandToken)
+                        {
+                            Console.WriteLine(startOfDataToken);
+                            CheckFeatureFlag();
+                        }
+                        Console.WriteLine(endOfDataToken);
+                    }
+
+                }
+                catch (Exception e) when (e is IOException || e is OutOfMemoryException || e is ArgumentOutOfRangeException)
+                {
+                    // Exit process
```

Comment:
We should get here when the main process exit and the pipe closes. We could be too restrictive here. Can you temporary comment out `when` and see if that helps?

Reply:
tested this on both sides (dynamo and the cli program), no difference.

In my latest commit I've implemented a simple watchdog on a timer as suggested by @pinzart90 - this works in my testing.